src/pipecat/services/anthropic.py

```diff
@@ -143,3 +198,265 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
 
         if context:
             await self._process_context(context)
+
+    async def request_image_frame(self, user_id: str, *, text_content: str = None):
+        await self.push_frame(UserImageRequestFrame(user_id=user_id, context=text_content), FrameDirection.UPSTREAM)
+
+    def _estimate_tokens(self, text: str) -> int:
+        return int(len(re.split(r'[^\w]+', text)) * 1.3)
+
+    async def _report_usage_metrics(self, prompt_tokens: int, completion_tokens: int):
+        if prompt_tokens or completion_tokens:
+            tokens = {
+                "processor": self.name,
+                "model": self._model,
+                "prompt_tokens": prompt_tokens,
+                "completion_tokens": completion_tokens,
+                "total_tokens": prompt_tokens + completion_tokens
+            }
+            await self.start_llm_usage_metrics(tokens)
```

Comment:
Maybe we can move this to `LLMService` (we would need to pass the model as an argument) so all LLMServices

Reply:
It's great to move things to parent classes if we can. The usage is reported differently by different services, so some usage aggregation/calculation maybe needs to be in the child classes. Or maybe I need to understand @mattieruth's aggregation code better.